File: Dot examples/ASP Book Examples/Book Examples/Chapter34/vb/App_Code/Entities/BlogTag.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Linq.Mapping
Imports System.Linq

''' <summary>
''' Summary description for BlogTag
''' </summary>
Partial Public Class BlogTag
     Inherits EntityBase(Of BlogTag)

    Public Shared Function GetSuggestions(ByVal prefixText As String, ByVal count As Integer) As String()
        Return Table.Where(Function(t) t.Name.StartsWith(prefixText)).Select(Function(t) t.Name).Distinct().Take(count).ToArray()
    End Function


    Public Function SelectByBlogId(ByVal blogId As Integer) As IEnumerable(Of BlogTag)
        Return Table.Where(Function(c) c.BlogId = blogId).OrderBy(Function(c) c.Name)
    End Function



    Protected Overrides Sub Validate()
        If Not ValidationUtility.SatisfiesRequired(Name) Then
            ValidationErrors.Add("Name", "Tag name is required")
        End If
    End Sub
End Class

File: Dot examples/ASP Book Examples/Book Examples/Chapter34/cs/App_Code/Entities/BlogTag.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Linq.Mapping;
using System.Linq;

/// <summary>
/// Summary description for BlogTag
/// </summary>
public partial class BlogTag : EntityBase<BlogTag>
{
    public static string[] GetSuggestions(string prefixText, int count)
    {
        return Table.Where( t => t.Name.StartsWith(prefixText) ).Select(t => t.Name).Distinct().Take(count).ToArray();
    }


    public IEnumerable<BlogTag> SelectByBlogId(int blogId)
    {
        return Table.Where(c => c.BlogId == blogId).OrderBy( c => c.Name); 
    }



    protected override void Validate()
    {
        if (!ValidationUtility.SatisfiesRequired(Name))
            ValidationErrors.Add("Name", "Tag name is required");
    }
}
